File: web/src/services/sys/sysDepartmentController.ts
// @ts-ignore
/* eslint-disable */
import { request } from '@umijs/max';

/** 此处后端没有提供注释 POST /api/sys/department/create */
export async function createSysDepartment(
  body: API.SysDepartmentCreateInput,
  options?: { [key: string]: any },
) {
  return request<API.RSysDepartmentDetailView>('/api/sys/department/create', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 DELETE /api/sys/department/delete/${param0} */
export async function deleteSysDepartmentById(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.deleteSysDepartmentByIdParams,
  options?: { [key: string]: any },
) {
  const { id: param0, ...queryParams } = params;
  return request<API.RBoolean>(`/api/sys/department/delete/${param0}`, {
    method: 'DELETE',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/sys/department/id/${param0} */
export async function findSysDepartmentById(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.findSysDepartmentByIdParams,
  options?: { [key: string]: any },
) {
  const { id: param0, ...queryParams } = params;
  return request<API.RSysDepartmentDetailView>(`/api/sys/department/id/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/sys/department/list */
export async function findSysDepartmentList(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.findSysDepartmentListParams,
  options?: { [key: string]: any },
) {
  return request<API.RListSysDepartmentPageView>('/api/sys/department/list', {
    method: 'GET',
    params: {
      ...params,
      specification: undefined,
      ...params['specification'],
    },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/sys/department/page */
export async function findSysDepartmentPage(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.findSysDepartmentPageParams,
  options?: { [key: string]: any },
) {
  return request<API.RListSysDepartmentPageView>('/api/sys/department/page', {
    method: 'GET',
    params: {
      ...params,
      specification: undefined,
      ...params['specification'],
      pageable: undefined,
      ...params['pageable'],
    },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/sys/department/tree */
export async function findSysDepartmentTree(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.findSysDepartmentTreeParams,
  options?: { [key: string]: any },
) {
  return request<API.RListSysDepartmentTreeView>('/api/sys/department/tree', {
    method: 'GET',
    params: {
      ...params,
      specification: undefined,
      ...params['specification'],
    },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/sys/department/tree-root */
export async function findSysDepartmentTreeRoot(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.findSysDepartmentTreeRootParams,
  options?: { [key: string]: any },
) {
  return request<API.RListSysDepartmentTreeRootView>('/api/sys/department/tree-root', {
    method: 'GET',
    params: {
      ...params,
      specification: undefined,
      ...params['specification'],
    },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /api/sys/department/update */
export async function updateSysDepartmentById(
  body: API.SysDepartmentUpdateInput,
  options?: { [key: string]: any },
) {
  return request<API.RSysDepartmentDetailView>('/api/sys/department/update', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

File: src/main/kotlin/com/zhengchalei/cloud/platform/modules/sys/controller/SysDepartmentController.kt
package com.zhengchalei.cloud.platform.modules.sys.controller

import com.zhengchalei.cloud.platform.commons.QueryPage
import com.zhengchalei.cloud.platform.commons.R
import com.zhengchalei.cloud.platform.modules.sys.domain.dto.*
import com.zhengchalei.cloud.platform.modules.sys.service.SysDepartmentService
import jakarta.validation.constraints.NotNull
import org.babyfish.jimmer.client.meta.Api
import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.*

@Api("sys")
@Validated
@RestController
@RequestMapping("/api/sys/department")
class SysDepartmentController(
    val sysDepartmentService: SysDepartmentService,
) {
    @PreAuthorize("hasAuthority('sys:department:id') or hasRole('admin')")
    @GetMapping("/id/{id}")
    fun findSysDepartmentById(
        @PathVariable id: Long,
    ): R<SysDepartmentDetailView> {
        val data = sysDepartmentService.findSysDepartmentById(id)
        return R(data = data)
    }

    @PreAuthorize("hasAuthority('sys:department:list') or hasRole('admin')")
    @GetMapping("/list")
    fun findSysDepartmentList(
        @NotNull specification: SysDepartmentPageSpecification,
    ): R<List<SysDepartmentPageView>> {
        val data = sysDepartmentService.findSysDepartmentList(specification)
        return R(data = data)
    }

    @PreAuthorize("hasAuthority('sys:department:tree-root') or hasRole('admin')")
    @GetMapping("/tree-root")
    fun findSysDepartmentTreeRoot(specification: SysDepartmentPageSpecification): R<List<SysDepartmentTreeRootView>> {
        val data = sysDepartmentService.findSysDepartmentTreeRoot(specification)
        return R(data = data)
    }

    @PreAuthorize("hasAuthority('sys:department:page') or hasRole('admin')")
    @GetMapping("/page")
    fun findSysDepartmentPage(
        @NotNull specification: SysDepartmentPageSpecification,
        @NotNull pageable: QueryPage,
    ): R<MutableList<SysDepartmentPageView>> {
        val data = sysDepartmentService.findSysDepartmentPage(specification, pageable.page())
        return R.success(data)
    }

    @PreAuthorize("hasAuthority('sys:department:tree') or hasRole('admin')")
    @GetMapping("/tree")
    fun findSysDepartmentTree(specification: SysDepartmentPageSpecification): R<List<SysDepartmentTreeView>> {
        val data = sysDepartmentService.findSysDepartmentTree(specification)
        return R(data = data)
    }

    @PreAuthorize("hasAuthority('sys:department:tree-select') or hasRole('admin')")
    @GetMapping("/tree-select")
    fun findSysDepartmentTreeSelect(specification: SysDepartmentPageSpecification): R<List<SysDepartmentTreeSelectView>> {
        val data = sysDepartmentService.findSysDepartmentTreeSelect(specification)
        return R(data = data)
    }

    @PreAuthorize("hasAuthority('sys:department:create') or hasRole('admin')")
    @PostMapping("/create")
    fun createSysDepartment(
        @NotNull @RequestBody sysDepartmentCreateInput: SysDepartmentCreateInput,
    ): R<SysDepartmentDetailView> {
        val data = sysDepartmentService.createSysDepartment(sysDepartmentCreateInput)
        return R(data = data)
    }

    @PreAuthorize("hasAuthority('sys:department:update') or hasRole('admin')")
    @PostMapping("/update")
    fun updateSysDepartmentById(
        @NotNull @RequestBody sysDepartmentUpdateInput: SysDepartmentUpdateInput,
    ): R<SysDepartmentDetailView> {
        val data = sysDepartmentService.updateSysDepartmentById(sysDepartmentUpdateInput)
        return R(data = data)
    }

    @PreAuthorize("hasAuthority('sys:department:delete') or hasRole('admin')")
    @DeleteMapping("/delete/{id}")
    fun deleteSysDepartmentById(
        @NotNull @PathVariable id: Long,
    ): R<Boolean> {
        sysDepartmentService.deleteSysDepartmentById(id)
        return R()
    }
}
